multicast/src/main/kotlin/com/dropbox/flow/multicast/Multicaster.kt

```diff
@@ -84,32 +84,43 @@ class Multicaster<T>(
     }
 
     /**
-     * The shared downstream flow. Collectors of this flow will share values dispatched by the
-     * [source] Flow.
+     * Gets a new downstream flow. Collectors of this flow will share values dispatched by a
+     * single upstream [source] Flow.
+     *
+     * @param piggybackOnly if true this downstream will cause a new upstream to start running
+     * (which only happens if no upstream is running alreay, e.g on if this is the first downstream
+     * added). Unlike a "regular" downstream, a [piggybackOnly] downstream can be closed without
+     * having any values emitted on it. [piggybackOnly] is only valid if [piggybackingDownstream] is
+     * enabled for this [Multicaster].
      */
-    val flow = flow<T> {
-        val channel = Channel<ChannelManager.Message.Dispatch.Value<T>>(Channel.UNLIMITED)
-        val subFlow = channel.consumeAsFlow()
-            .onStart {
-                try {
-                    channelManager.addDownstream(channel)
-                } catch (closed: ClosedSendChannelException) {
-                    // before we could start, channel manager was closed.
-                    // close our downstream manually as it won't be closed by the ChannelManager
-                    channel.close()
+    fun newDownsteam(piggybackOnly: Boolean = false): Flow<T> {
```

Comment:
hmm i don't like this much as it is no more a field but a creator function now but does not matter that much.

Reply:
I played around with adding two fields but didn't like how it looked. Given that the code inside `flow` execute on every collect anyway I don't think it's a big deal so going with the clearer approach